### ISSUE TITLE
why optimizer.zero_grad() after optimizer.step()?

### ISSUE THREAD

**Issue by littleWangyu** [inyeoplee77/SAGPool#15]
in train epoch:
why optimizer.zero_grad() after optimizer.step()?
Does it matter?
It's usually optimizer.step()--loss.backward()--optimizer.step()
